File: int_tech/ci/coursework/normalise/analysis.m
num_files = 6;
num_cols = 17;
tst_sizes = [];
trn_sizes = [];
for i = 1:num_files;
    tst_sizes(i, :) = size(load(strcat(int2str(i), 'tst.norm.ssv')));
    trn_sizes(i, :) = size(load(strcat(int2str(i), 'trn.norm.ssv')));
end

tst = [];
trn = [];
for i = 1:num_files;
    tst(i,:,:) = load(strcat(int2str(i), 'tst.norm.ssv'));
    trn(i,:,:) = load(strcat(int2str(i), 'trn.norm.ssv'));
end;

data_mins = [];
data_maxs = [];
data_means = [];
for i = 1:num_files;
    for j = 1:num_cols;
        data_means(i, j)             = mean(tst(i, :, j));
        data_means(num_files + i, j) = mean(trn(i, :, j));
        data_mins(i, j)              = min(tst(i, :, j));
        data_mins(num_files + i, j)  = min(trn(i, :, j));
        data_maxs(i, j)              = max(tst(i, :, j));
        data_maxs(num_files + i, j)  = max(trn(i, :, j));
    end
end

File: int_tech/ci/coursework/normalise/analysis.R
# The R version for the analysis.m file (it runs faster)

filens <- c("trn.ssv", "tst.ssv")
frames <- c(1,2,3,4,5,6)
fmat   <- sapply(filens, function(x) { paste(frames, x, sep="") })
nms    <- c(fmat[,1], fmat[,2])
path   <- "../third_party/"
flnm   <- function(x) { paste(path, x, sep="") }
files  <- lapply(nms, function(x) { sapply(x, flnm) })
names(files) <- nms
data   <- lapply(nms, function(x) read.table(files[[x]]))
names(data)  <- nms
trnsz  <- sapply(fmat[,1], function(x) { dim(data[[x]]) })
tstsz  <- sapply(fmat[,2], function(x) { dim(data[[x]]) })
app    <- function(f) {
    Reduce(function(x, y) { rbind(x, apply(y, 2, f)) }, data, NULL)
}
mmm    <- lapply(c(mean, min, max), app)
names(mmm)   <- c('mean', 'min', 'max')
cat("Sizes:\n")
print(trnsz)
print(tstsz)
cat("\nStats:\n")
print(mmm)

